Authentication_and_Authorization/hello-security/signup-controller.php

```diff
@@ -1,20 +1,63 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <title>Login</title>
+</head>
+<body>
+<a href="index.php"> Take me home, country roads ... </a> <br>
+
 <?php
 include "db-connection.php";
 
-$conn = openCon();
+function checkIfEmailExist($email){
+  $conn = openCon();
 
-$sql = "INSERT INTO person (email, firstname, password, role) values ('"
-  . htmlentities($_POST["email"]) . "', '" 
-  . htmlentities($_POST["firstname"]). "', '"
-  . htmlentities($_POST["password"]) . "', '"
-  . htmlentities($_POST["role"]) . "');";
-// echo ("sql: " . $sql);  
-if (!$conn -> query($sql)) {
-  echo("Error description: " . $conn -> error);
-} else {
-	echo("You are registered as: " . $_POST["email"] . " with role: " .  $_POST["role"]);
+  $sql = "SELECT * FROM person where email = :email;";
+  $preparedSql = $conn->prepare($sql);
+  $preparedSql->bindParam(':email',$email);
+  $preparedSql->execute() or die("Failed to check is email exist.");
+  
+  $result = $preparedSql->fetchAll();
+  
+  return count($result) !== 0;
 }
-?>
 
+$email = htmlentities($_POST["email"]);
+$firstname = htmlentities($_POST["firstname"]);
+$password = htmlentities($_POST["password"]);
+$role = htmlentities($_POST["role"]);
+ 
+if( !empty($email) && !empty($firstname) && !empty($password) && !empty($role)){
+
+  if(strcmp($role, "admin")){
```

Comment:
I am able to register with a custom role, e.g. "root". Currently I cannot do anything specifically with that role currently but this could be considered a security issue.